### PR TITLE
AnnouncementViewModel Test

### PR DESCRIPTION
**Pull request by Mzazi25**
Unit tests for AnnouncementViewModel

### CODE REVIEW

File: app/src/test/java/org/greenstand/android/TreeTracker/messages/announcementmessage/AnnouncementViewModelTest.kt [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+    @Before
+    fun setup(){
+        coEvery { messagesRepo.getAnnouncementMessages(messageId) } returns fakeAnnouncementMessage
+        announcementViewModel = AnnouncementViewModel(messageId, messagesRepo)
+    }
+
+    @Test
+    fun `verify messages repo gets the correct announcement message `()= runBlocking {
+        coEvery { messagesRepo.getAnnouncementMessages(messageId) } returns fakeAnnouncementMessage
```
Comment (Elforama) [inline app/src/test/java/org/greenstand/android/TreeTracker/messages/announcementmessage/AnnouncementViewModelTest.kt:38]: You already put this in the @Before method, so there is no need to put it in all the other methods. Remove this from all other methods.

**Review by Elforama — CHANGES_REQUESTED** [review]
Looks good, just some minor clean up needed.

File: app/src/test/java/org/greenstand/android/TreeTracker/messages/announcementmessage/AnnouncementViewModelTest.kt [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    @get:Rule
+    var mainCoroutineRule = MainCoroutineRule()
+
+    private val messageId = ""
+    private val messagesRepo = mockk<MessagesRepo>(relaxed = true)
+    private lateinit var announcementViewModel: AnnouncementViewModel
+
+    @Before
+    fun setup(){
+        coEvery { messagesRepo.getAnnouncementMessages(messageId) } returns fakeAnnouncementMessage
```
Comment (Elforama) [inline app/src/test/java/org/greenstand/android/TreeTracker/messages/announcementmessage/AnnouncementViewModelTest.kt:32]: This doesn't need a specific id. You can pass in `any()` instead. The only places to set the messageID is when creating the viewmodel

File: app/src/test/java/org/greenstand/android/TreeTracker/messages/announcementmessage/AnnouncementViewModelTest.kt (hunk 1 above)
Comment (Mzazi25) [reply app/src/test/java/org/greenstand/android/TreeTracker/messages/announcementmessage/AnnouncementViewModelTest.kt]: Done

File: app/src/test/java/org/greenstand/android/TreeTracker/messages/announcementmessage/AnnouncementViewModelTest.kt [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    fun `WHEN get announcement is triggered THEN current body changes`()= runBlocking {
+        coEvery { messagesRepo.getAnnouncementMessages(messageId) } returns fakeAnnouncementMessage
+        announcementViewModel.state.test {
+            assertEquals(awaitItem().currentBody, fakeAnnouncementMessage.body)
+        }
+    }
+
+    @Test
+    fun `verify message repo marks message as read`()= runBlocking {
+        coEvery { messagesRepo.markMessageAsRead(messageId) }
```
[the remark this answers is not in the document]
Comment (Mzazi25) [reply app/src/test/java/org/greenstand/android/TreeTracker/messages/announcementmessage/AnnouncementViewModelTest.kt]: Done

File: app/build.gradle [hunk 4]
```diff
@@ -271,6 +271,8 @@ dependencies {
     androidTestImplementation "com.android.support.test:runner:1.0.2"
     androidTestImplementation "com.android.support.test.espresso:espresso-core:3.0.2"
     testImplementation "org.robolectric:robolectric:4.8.1"
+    androidTestImplementation "app.cash.turbine:turbine:0.12.1"
+    testImplementation "app.cash.turbine:turbine:0.12.1"
```
[the remark this answers is not in the document]
Comment (Mzazi25) [reply app/build.gradle:275]: Okay. I will

File: app/src/test/java/org/greenstand/android/TreeTracker/messages/announcementmessage/AnnouncementViewModelTest.kt (hunk 2 above)
Comment (Mzazi25) [reply app/src/test/java/org/greenstand/android/TreeTracker/messages/announcementmessage/AnnouncementViewModelTest.kt]: Done

**Comment by Mzazi25** [comment]
Made the changes

File: app/src/test/java/org/greenstand/android/TreeTracker/messages/announcementmessage/AnnouncementViewModelTest.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (Elforama) [reply app/src/test/java/org/greenstand/android/TreeTracker/messages/announcementmessage/AnnouncementViewModelTest.kt]: You don't need to delete the method, just the coEvery part.

**Review by Elforama — APPROVED** [review]

**Pull request merged**